joblib/numpy_pickle.py

```diff
@@ -446,22 +336,35 @@ def load(filename, mmap_mode=None):
     file was saved with compression, the arrays cannot be memmaped.
     """
     with open(filename, 'rb') as file_handle:
+        magic = _read_magic(file_handle)
+
+    # Backward compatibility with old compression strategy
+    if magic == _ZFILE_PREFIX:
+        warnings.warn("You are using an old version of joblib cache. Please "
+                      "regenerate your cache.",
+                      DeprecationWarning, stacklevel=2)
+        return load_compatibility(filename)
+
+    with closing(_check_filetype(filename, magic)) as file_handle:
+        if isinstance(file_handle, gzip.GzipFile) and mmap_mode is not None:
+            raise ValueError('File "%(filename)s" appears to be compressed, '
+                             'this is not compatible with mmap_mode '
+                             '"%(mmap_mode)s" flag passed'
+                             % locals())
+
         # We are careful to open the file handle early and keep it open to
         # avoid race-conditions on renames. That said, if data are stored in
         # companion files, moving the directory will create a race when
```

Comment:
There are no companion files anymore (I am guessing they were the .npy files). I have to say I don't feel I understand this comment 100% but maybe just remove the second sentence.